### ISSUE TITLE
Issue in adding http package in extended components

### ISSUE THREAD

**Issue by r2ast** [akveo/nebular#2785]
<!--
What I am trying to do it, i created a customer login page which is extends the NbLoginComponent.

How when i try to implement the http package. it is not allowing me to do so. If i try to use the constructor, it is not also working. How can i use the http in extended components.

Thanks

**Comment by r2ast** [comment]
<!--
What I am trying to do it, i created a customer login page which is extends the NbLoginComponent.

How when i try to implement the http package. it is not allowing me to do so. If i try to use the contrctor, it is not also working. How can i use the http in extended components.

Thanks

**Comment by r2ast** [comment]
<!--
What I am trying to do it, i created a customer login page which is extends the NbLoginComponent.

How when i try to implement the http package. it is not allowing me to do so. If i try to use the contrctor, it is not also working. How can i use the http in extended components.

Thanks

**Comment by r2ast** [comment]
Hi fellows,

Lets me recreate the issue that I have faced. What exactly I am trying to do is that when i try to extent the NbLoginComponent which extents the libraries, theme etc. When i try to implement the constructor which inherits the base class( in this case NbLoginComponent ) in our customer NgxLoginComponent componets and try to use the argument from base class using super, it remove the inherits class features ( remove the templates and models).

All the css properties and alerts not working. Also i am unable to use the http package and also i am unable to make http  request to my customer services.

I try to read all the resources available but there is no exact point solution that i could found. 

Finally with lots of efforts i was able to fix the issue. Here is the solution:
1. In am customer login page . I added. a provider for the services. 
 Like this 

@Component({
  selector: 'ngx-login',
  templateUrl: './login.component.html',
  styleUrls: ['./login.component.scss'],
  providers: [
    {provide: SchoolDataService, useClass : SchoolDataService}]
}) 

2.  I added the constructor like this:
 export class NgxLoginComponent extends NbLoginComponent {
  techerList: any;

  constructor(private fb: FormBuilder, @Inject(NB_AUTH_OPTIONS) protected options: {}, service: NbAuthService, private http: HttpClient,public schoolDataService:SchoolDataService,
    cd: ChangeDetectorRef, router: Router) {
    super(service, options, cd, router)};
